src/python/pants/engine/scheduler.py

```diff
@@ -401,40 +386,24 @@ def execution_request(self, products, subjects):
       :class:`pants.engine.fs.PathGlobs` objects.
     :returns: An ExecutionRequest for the given products and subjects.
     """
-    return ExecutionRequest(tuple((s, p) for s in subjects for p in products))
-
-  def root_entries(self, execution_request):
-    """Returns the roots for the given ExecutionRequest as a list of tuples of:
-         ((subject, product), State)
-    """
-    with self._product_graph_lock:
-      if self._execution_request is not execution_request:
-        raise AssertionError(
-          "Multiple concurrent executions are not supported! {} vs {}".format(
-            self._execution_request, execution_request))
-      return self._scheduler.root_entries(execution_request)
+    roots = tuple((s, p) for s in subjects for p in products)
+    native_execution_request = self._scheduler._native.new_execution_request()
```

Comment:
Is it possible for this constructor to take a tuple of subjects and products, rather than needing to create one and mutate it?

Reply:
This is supposed to be a form of native/python-interop builder-pattern... at one point it was more necessary because there was more than one root type, and adding structs to represent all of them seemed like overkill.

But it's very possible we'll add back the ability to have multiple root types (more than just `Select(X)`), and it isn't adding much load, so I'd like to leave it if that is fine with you.